File: 016-mile-pace/milePace.js
/*
Pseudocode
1. reduce time to seconds and change to num data type
2. divide time by mile given
3. transform answeer to minutes:seconds format and return as answer
*/

function milePace(miles, time) {
  let avgTime, newMin, newSec;

  if (miles === 1) {
    avgTime = time
  } else {
    const timeArr = time.split(":")
    let timeToSec = (Number(timeArr[0]) * 60) + Number(timeArr[1]);
    newMin = String(Math.floor((timeToSec / miles) / 60));
    newSec = String(Math.round((timeToSec / miles) % 60));
    if (newMin.length < 2) {
      newMin = "0" + newMin;
    }

    if (newSec < 2) {
      newSec = newSec + "0";
    }

    avgTime = newMin + ":" + newSec;
  }

  return avgTime;
}

console.log(milePace(3, "24:00")); // should return "08:00"
console.log(milePace(1, "06:45")); // should return "06:45"
console.log(milePace(2, "07:00")); // should return "03:30"
console.log(milePace(26.2, "120:35")); // should return "04:36"

File: 016-mile-pace/mile_pace.py
import math

'''
1. Split and transform input time to number format (in seconds)
2. Calculate total time (in sec) divided by mile to get avg minutes
3. Calculate total time (in sec) modulo mile to get avg seconds
Remember to round down both values to nearest whole number and transform back to string after calc
4. Check if the min or sec str len is 1, if yes, pad with 0 appropriately
5. Combine and return final str
'''

def mile_pace(mile, time):
  avg_time = ""

  if mile == 1:
    avg_time = time
  else:
    time_list = time.split(":")
    time_to_sec = (int(time_list[0]) * 60) + int(time_list[1])
    avg_min = "0" + str(math.floor((time_to_sec // mile) // 60)) if len(str(math.floor((time_to_sec // mile) // 60))) == 1 else str(math.floor((time_to_sec // mile) // 60))
    avg_sec = str(math.floor((time_to_sec // mile) % 60)) + "0" if len(str(math.floor((time_to_sec // mile) % 60))) == 1 else str(math.floor((time_to_sec // mile) % 60))
    avg_time = avg_min + ":" + avg_sec

  return avg_time